scala/x509.scala

```diff
@@ -0,0 +1,38 @@
+import org.mongodb.scala._
+import scala.collection.JavaConverters._
+import com.mongodb.connection.ClusterSettings
+import com.mongodb.connection.SslSettings
+import org.mongodb.scala.Document
+import org.mongodb.scala.bson._
+
+object MongoDBx509 extends App {
+   System.setProperty("javax.net.ssl.trustStore","client.truststore");
+   System.setProperty("javax.net.ssl.trustStorePassword","password");
+
+   System.setProperty("javax.net.ssl.keyStore", "client.keystore");
+   System.setProperty("javax.net.ssl.keyStorePassword", "password");
+
+   // null value will force mongoDB server to use DN from cert
+   val credential = MongoCredential.createMongoX509Credential(null);
+
+   val settings = MongoClientSettings.builder()
+      .credential(credential)
+      .applyToClusterSettings(
+         (builder: ClusterSettings.Builder)
+         => builder.hosts(List(new ServerAddress("localmongo1", 27017)).asJava))
+      .applyToSslSettings(
+         (builder : SslSettings.Builder)
+         => builder.enabled(true))
+      .build();
+
+   val client : MongoClient = MongoClient(settings);
+   
+   val listDatabaseNames = client.listDatabaseNames();
+
+   listDatabaseNames.foreach(println)
+
+   val db : MongoDatabase = client.getDatabase("test")
+   val collection : MongoCollection[Document] = db.getCollection("stuff")
+   collection.insertOne(Document("scalatest" -> BsonString("OK"))).subscribe((results: Completed) => println(results))
+   collection.find().collect().subscribe((results: Seq[Document]) => println(s"Found: #${results.size}"))
```

Comment:
I think the other snippets return the id of the newly-inserted document. Could we change this to the id rather than number of docs?